File: nysol_python/mcmd/methods/convTex/db/smp/mselrand.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='mselrand'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
customer,date,amount
A,20081201,10
A,20081207,20
A,20081213,30
B,20081002,40
B,20081209,50
'''
db['idatas'].append(data)

data={}
data['name']='dat2.csv'
data['text']='''
customer,date,amount
A,20081201,10
A,20081207,20
A,20081213,30
B,20081002,40
B,20081209,50
C,20081210,60
D,20081201,70
D,20081205,80
D,20081209,90
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
一人の顧客につきランダムに1行を選択する。
'''
script['sh_code']='''
mselrand k=顧客 c=1 S=1 i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mselrand(k="customer", c="1", S="1", i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='ランダムに一定割合の行を選択'
script['comment']='''
一人の顧客につきランダムに50\%の行を選択する。
また、それ以外の不一致データは ``oth2.csvと`` いうファイルに出力する。
'''
script['sh_code']='''
mselrand k=顧客 p=50 S=1 u=oth2.csv i=dat1.csv o=rsl2.csv
'''
script['py_code']='''
nm.mselrand(k="customer", p="50", S="1", u="oth2.csv", i="dat1.csv", o="rsl2.csv").run()
'''
script['odatas']='oth2.csv,rsl2.csv'
db['scripts'].append(script)

script={}
script['title']='キー単位の選択'
script['comment']='''
以下の例は、顧客 ``A,B,C,D`` の4人からランダムに2人を選ぶ。
顧客 ``D`` が選ばれると、顧客 ``D`` の行は全て出力される。
'''
script['sh_code']='''
mselrand k=顧客 c=2 S=1 -B i=dat2.csv o=rsl3.csv
'''
script['py_code']='''
nm.mselrand(k="customer", c="2", S="1", B=True, i="dat2.csv", o="rsl3.csv").run()
'''
script['odatas']='rsl3.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/mselrand.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
Customer,Date,Amount
A,20081201,10
A,20081207,20
A,20081213,30
B,20081002,40
B,20081209,50
EOF
)}

File.open("dat2.csv","w"){|fpw| fpw.write(
<<'EOF'
Customer,Date,Amount
A,20081201,10
A,20081207,20
A,20081213,30
B,20081002,40
B,20081209,50
C,20081210,60
D,20081201,70
D,20081205,80
D,20081209,90
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
Randomly select 1 transaction for each customer.
EOF
scp=<<'EOF'
more dat1.csv
mselrand k=Customer c=1 S=1 i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## Example 2
title="Randomly select a percentage of records"
comment=<<'EOF'
Select 50\% of each customers' records at random. Save other records to a separate file \verb|oth.csv|.
EOF
scp=<<'EOF'
mselrand k=Customer p=50 S=1 u=oth2.csv i=dat1.csv o=rsl2.csv
more rsl2.csv
more oth2.csv
EOF
run(scp,title,comment)

############## Example 3
title="Select records by same key"
comment=<<'EOF'
In the following example, select two out of the four customers \verb|A,B,C,D| at random.
Customer \verb|C,D| is selected, and all records of customer \verb|C,D| is printed to the output.

EOF
scp=<<'EOF'
more dat2.csv
mselrand k=Customer c=2 S=1 -B i=dat2.csv o=rsl3.csv
more rsl3.csv
EOF
run(scp,title,comment)
